src/OpenTelemetry.Collector.Dependencies/AzureSdkDiagnosticListener.cs

```diff
@@ -20,20 +20,17 @@ namespace OpenTelemetry.Collector.Dependencies
     using System.Collections.Generic;
     using System.Diagnostics;
     using System.Net.Http;
-    using OpenTelemetry.Collector.Dependencies.Implementation;
+    using OpenTelemetry.Collector.Dependencies.Implementation;
     using OpenTelemetry.Trace;
 
     internal class AzureSdkDiagnosticListener : ListenerHandler<HttpRequestMessage>
     {
         private static readonly PropertyFetcher LinksPropertyFetcher = new PropertyFetcher("Links");
-        private readonly ITracer tracer;
-
         private readonly ISampler sampler;
 
-        public AzureSdkDiagnosticListener(string sourceName, ITracer tracer, ISampler sampler)
-            : base(sourceName, tracer, null)
+        public AzureSdkDiagnosticListener(string sourceName, Version version, ITracerFactory tracerFactory, ISampler sampler)
+            : base(sourceName, version, tracerFactory, null)
```

Comment:
same as HttpInListener: this should deal with specifc tracer rather than factory and you can request a tracer in subscriber